File: app/src/test/java/com/omar/retromp3recorder/app/usecase/ChangeSampleRateUCImplTest.java
package com.omar.retromp3recorder.app.usecase;

import com.omar.retromp3recorder.app.di.DaggerTestAppComponent;
import com.omar.retromp3recorder.app.recorder.VoiceRecorder;
import com.omar.retromp3recorder.app.repo.SampleRateRepo;

import org.junit.Before;
import org.junit.Test;

import javax.inject.Inject;

public class ChangeSampleRateUCImplTest {

    @Inject
    SampleRateRepo sampleRateRepo;
    private ChangeSampleRateUCImpl changeSampleRateUC;

    @Before
    public void setUp() {
        DaggerTestAppComponent.create().inject(this);
        changeSampleRateUC = new ChangeSampleRateUCImpl(sampleRateRepo);
    }

    @Test
    public void test_UC_Executed() {
        changeSampleRateUC.execute(VoiceRecorder.SampleRate._8000).subscribe();

        //Then
        sampleRateRepo.observe().test().assertValue(
          VoiceRecorder.SampleRate._8000
        );
    }

}

File: app/src/test/java/com/omar/retromp3recorder/app/usecases/ChangeSampleRateUCImplTest.kt
package com.omar.retromp3recorder.app.usecases

import com.omar.retromp3recorder.app.di.DaggerTestAppComponent
import com.omar.retromp3recorder.app.state.SampleRateRepo
import com.omar.retromp3recorder.recorder.Mp3VoiceRecorder
import org.junit.Before
import org.junit.Test
import javax.inject.Inject

class ChangeSampleRateUCImplTest {
    @Inject
    lateinit var sampleRateRepo: SampleRateRepo
    private lateinit var changeSampleRateUC: ChangeSampleRateUC
    @Before
    fun setUp() {
        DaggerTestAppComponent.create().inject(this)
        changeSampleRateUC = ChangeSampleRateUC(sampleRateRepo)
    }

    @Test
    fun `assert happy flow`() {
        changeSampleRateUC.execute(Mp3VoiceRecorder.SampleRate._8000).subscribe()

        //Then
        sampleRateRepo.observe().test().assertValue(
            Mp3VoiceRecorder.SampleRate._8000
        )
    }
}